### PR TITLE
PWN-6320 - Add Moonpay Sanbox switch to debug menu

### PR DESCRIPTION
**Pull request by davran1001**
Screenshot 2022-12-13 at 19 40 46

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsAdapter.kt [hunk 1]
```diff
@@ -134,4 +139,32 @@ class DebugSettingsAdapter(
...
+        binding: ItemSettingsSwitchBinding,
+        private val onSettingsRowSwitchListener: (titleResId: Int, isSelected: Boolean) -> Unit
+    ) : RecyclerView.ViewHolder(binding.root) {
+
+        private val title = binding.textViewSettingName
+        private val subtitle = binding.textViewSettingSubtitle
+        private val switcher = binding.switchChangeSettings
+        private val icon = binding.imageViewSettingIcon
+
+        constructor(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsAdapter.kt:153]: can be done more compat using parent and inflateViewBinding

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsContract.kt [hunk 2]
```diff
@@ -14,5 +14,6 @@ interface DebugSettingsContract {
     interface Presenter : MvpPresenter<View> {
         fun loadData()
         fun onNetworkChanged(newNetworkEnvironment: NetworkEnvironment)
+        fun updateMoonpayUrl(isSandboxSelected: Boolean)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsContract.kt:17]: switchMoonpayUrl

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsFragment.kt [hunk 3]
```diff
@@ -92,6 +92,14 @@ class DebugSettingsFragment :
         }
     }
 
+    private fun onSettingsSwitchClicked(@StringRes titleResId: Int, isSelected: Boolean) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsFragment.kt:95]: i would stick with the isChecked

File: app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkServicesUrlProvider.kt [hunk 4]
```diff
@@ -91,14 +91,27 @@ class NetworkServicesUrlProvider(
...
-    fun saveMoonpayEnvironment(newServerSideUrl: String) {
+    fun toggleMoonpayEnvironment(isSandboxSelected: Boolean) {
+        if (isSandboxSelected) {
+            saveMoonpayEnvironment(context.getString(R.string.moonpayServerSideProxySandboxUrl))
+        } else {
+            saveMoonpayEnvironment(context.getString(R.string.moonpayServerSideProxyUrl))
+        }
+    }
+
+    fun isMoonPaySandboxActive(): Boolean {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkServicesUrlProvider.kt:109]: isMoonpaySandboxActive

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkServicesUrlProvider.kt:109]: or i would just extracted a flag inside MoonpayEnvironment `isSandbox`

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsFragment.kt (hunk 3 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsFragment.kt:95]: that's what I talked about  x D 20 minutes

**Pull request merged**